more mutable fixes + some extra cleanups

bgruening opened the pull request.

bgruening committed:
rework some more mutable arguments

bgruening committed:
rework some more mutable arguments

bgruening committed:
rework some more mutable arguments

bgruening committed:
use str(e) instead of e.message

bgruening committed:
remove some whitespaces and enumerate

bgruening committed:
remove mutable function arguments

bgruening committed:
remove mutable function arguments

bgruening committed:
remove mutable function arguments

bgruening committed:
use ValueError instead of assert False

bgruening committed:
replace mutable argument

bgruening committed:
small fix

bgruening committed:
replace mutable argument

bgruening committed:
use values instead of items()

bgruening committed:
add a few more fixes all over the place

bgruening committed:
improve some dict query

bgruening committed:
more mutable argument fixes

bgruening committed:
more mutable fixes + items --> values

bgruening committed:
python3 fix

bgruening committed:
replace setattr with more pythonic code

bgruening committed:
items --> values

bgruening committed:
prefix unused vars

bgruening committed:
items() --> values()

bgruening committed:
prefix unused vars

bgruening committed:
items --> keys

bgruening committed:
remove enumerate

bgruening committed:
remove enumerate and mutable argument

bgruening committed:
remove enumerate, prefix unused vars

bgruening committed:
prefix unused vars

bgruening committed:
f-strings

bgruening committed:
prefix unused var

bgruening committed:
getattr --> pythonic

bgruening committed:
mutable args

bgruening committed:
items --> values

bgruening committed:
prefix unused vars

bgruening committed:
mutatable arguments

bgruening committed:
more fixes

bgruening committed:
add more fixes to ./lib/tool_shed

nsoranzo commented on lib/galaxy/authnz/managers.py:
```diff
@@ -313,8 +313,8 @@ def create_user(self, provider, token, trans, login_redirect_url):
-            log.exception(e.message)
-            raise exceptions.AuthenticationFailed(e.message)
+            log.exception(str(e))
+            raise exceptions.AuthenticationFailed(str(e))
```
```suggestion
            log.exception("Error creating user")
            raise
```

nsoranzo commented on lib/galaxy/selenium/navigates_galaxy.py:
```diff
@@ -1217,7 +1217,7 @@ def history_multi_view_display_collection_contents(self, collection_hid, collect
         self.click(selector)
         next_level_element_selector = selector
-        for i in range(len(collection_type.split(":")) - 1):
+        for _i in range(len(collection_type.split(":")) - 1):
```
```suggestion
        for _ in range(len(collection_type.split(":")) - 1):
```

bgruening committed:
fix mutable arguments in tests

bgruening committed:
small fixes

nsoranzo commented on lib/tool_shed/util/search_util.py:
```diff
@@ -57,10 +57,10 @@ def in_workflow_dict(workflow_dict, exact_matches_checked, workflow_name):
     # If either list is 1 item, we'll append to it until its length is the same as the other.
     if len(list1) == 1:
-        for i in range(1, len(list2)):
+        for _i in range(1, len(list2)):
```
```suggestion
        for _ in range(1, len(list2)):
```

nsoranzo commented on lib/tool_shed/util/search_util.py:
```diff
@@ -57,10 +57,10 @@ def in_workflow_dict(workflow_dict, exact_matches_checked, workflow_name):
             list1.append(list1[0])
     elif len(list2) == 1:
-        for i in range(1, len(list1)):
+        for _i in range(1, len(list1)):
```
```suggestion
        for _ in range(1, len(list1)):
```

nsoranzo commented on lib/tool_shed/test/base/twilltestcase.py:
```diff
@@ -528,8 +533,10 @@ def deactivate_repository(self, installed_repository, strings_displayed=None, st
+    def delete_files_from_repository(self, repository, filenames=None, strings_displayed=None, strings_not_displayed=None):
+        filenames = filenames or []
         files_to_delete = []
+        strings_displayed = strings_displayed or ['were deleted from the repository']
```
```suggestion
        if strings_displayed is None:
            strings_displayed = ['were deleted from the repository']
```

nsoranzo commented on lib/tool_shed/grids/repository_review_grids.py:
```diff
@@ -92,7 +92,8 @@ def get_value(self, trans, grid, repository):
                     rval += '%s</a> | ' % user.username
-                rval = rval.rstrip(' | ')
+                if rval[-3:] == ' | ':
+                    rval = rval[:-3])
```
```suggestion
                    rval = rval[:-3]
```

nsoranzo commented on lib/galaxy/webapps/galaxy/api/display_applications.py:
```diff
@@ -24,17 +24,17 @@ def index(self, trans, **kwd):
                 'version': display_app.version,
                 'filename_': display_app._filename,
-                'links': [{'name': l.name} for l in display_app.links.values()]
+                'links': [{'name': _.name} for _ in display_app.links.values()]
```
```suggestion
                'links': [{'name': link.name} for link in display_app.links.values()]
```

nsoranzo commented on lib/galaxy/tool_util/deps/brew_exts.py:
```diff
@@ -494,7 +495,7 @@ def versioned(recipe_path):
     # TODO: Also use tags.
     stdout = brew_execute(["versions", package])
-    version_parts = [l for l in stdout.split("\n") if l and "git checkout" in l]
+    version_parts = [_ for _ in stdout.split("\n") if _ and "git checkout" in _]
```
```suggestion
    version_parts = [line for line in stdout.split("\n") if line and "git checkout" in line]
```

nsoranzo commented on lib/galaxy/tool_util/deps/brew_exts.py:
```diff
@@ -51,7 +51,7 @@ class BrewContext:
         ensure_brew_on_path(args)
         raw_config = brew_execute(["config"])
-        config_lines = [l.strip().split(":", 1) for l in raw_config.split("\n") if l]
+        config_lines = [_.strip().split(":", 1) for _ in raw_config.split("\n") if _]
```
```suggestion
        config_lines = [line.strip().split(":", 1) for line in raw_config.split("\n") if line]
```

nsoranzo commented on lib/galaxy/tool_shed/util/container_util.py:
```diff
@@ -60,9 +60,9 @@ def get_components_from_key(key):
+    for _i in range(1, pad):
         pad_str += ' '
-    print('{}id: {} key: {}'.format(pad_str, str(folder.id), folder.key))
+    print(f'{pad_str} id: {folder.id} key: {folder.key}')
```
```suggestion
    print(f'{pad_str}id: {folder.id} key: {folder.key}')
```

nsoranzo commented on lib/galaxy/tool_shed/util/container_util.py:
```diff
@@ -60,9 +60,9 @@ def get_components_from_key(key):
 def print_folders(pad, folder):
     # For debugging...
     pad_str = ''
```
```suggestion
    pad_str = ' ' * pad
```

nsoranzo commented on lib/galaxy/tool_shed/util/container_util.py:
```diff
@@ -60,9 +60,9 @@ def get_components_from_key(key):
     pad_str = ''
-    for i in range(1, pad):
+    for _i in range(1, pad):
         pad_str += ' '
```
Not sure why `range` started from 1, but it was missing one space.
```suggestion
```

nsoranzo commented on lib/galaxy/tool_shed/tools/data_table_manager.py:
```diff
@@ -145,7 +145,7 @@ def install_tool_data_tables(self, tool_shed_repository, tool_index_sample_files
                     if elem.tag == 'tables':
-                        for table_elem in elems:
+                        for _table_elem in elems:
                             elems.append(elem)
```
This code doesn't make much sense to me.

nsoranzo commented on lib/galaxy/tool_shed/galaxy_install/tools/tool_panel_manager.py:
```diff
@@ -467,9 +468,9 @@ def update_tool_panel_dict(self, tool_panel_dict, tool_panel_section_mapping, re
             if tool_guid not in tool_panel_section_mapping:
                 continue
-            for idx, tool in enumerate(tool_panel_dict[tool_guid]):
+            for tool in enumerate(tool_panel_dict[tool_guid]):
```
```suggestion
            for tool in tool_panel_dict[tool_guid]:
```

nsoranzo commented on lib/galaxy/tool_shed/galaxy_install/tool_migration_manager.py:
```diff
@@ -412,7 +412,7 @@ def handle_repository_contents(self, tool_shed_repository, repository_clone_url,
                     for k, v in tool_panel_dict_for_tool_config.items():
                         tool_panel_dict_for_display[k] = v
-                        for tool_panel_dict in v:
+                        for _tool_panel_dict in v:
                             # Keep track of tool config file names associated with entries that have been made to the
                             # migrated_tools_conf.xml file so they can be eliminated from all non-shed-related tool panel configs.
                             if tool_config not in tool_configs_to_filter:
```
This code doesn't make much sense to me.

bgruening replied:
Should I add some TODOs? There are some cases that are wired and probably never worked, but I feel this is not the correct PR to fix it.

nsoranzo replied:
Adding a TODO is fine by me, happy I'm not the only one finding them weird.

nsoranzo commented on lib/galaxy/tool_shed/util/container_util.py:
```diff
@@ -60,9 +60,9 @@ def get_components_from_key(key):
 def print_folders(pad, folder):
```
This method seems to be unused, so it could actually make sense to remove it from here and lib/tool_shed/util/container_util.py

bgruening replied:
Done.

jmchilton commented on test/unit/workflows/test_workflow_progress.py:
```diff
@@ -218,7 +218,8 @@ class MockModuleInjector:
 
-    def inject(self, step, step_args={}):
+    def inject(self, step, step_args=None):
+        step_args = step_args or {}
```
Looks like you don't need this line.

bgruening replied:
Done.

bgruening committed:
Apply suggestions from code review

Co-authored-by: Nicola Soranzo <nicola.soranzo@earlham.ac.uk>

bgruening committed:
remove print_folders() function

bgruening committed:
remove print_folders() function

bgruening committed:
adopt to code review

nsoranzo commented on lib/galaxy/authnz/managers.py:
```diff
@@ -313,8 +313,8 @@ def create_user(self, provider, token, trans, login_redirect_url):
                 return False, message, (None, None)
             return success, message, backend.create_user(token, trans, login_redirect_url)
         except exceptions.AuthenticationFailed as e:
```
```suggestion
        except exceptions.AuthenticationFailed:
```

nsoranzo commented on lib/tool_shed/test/base/twilltestcase.py:
```diff
@@ -174,7 +176,7 @@ def submit_form(self, form_no=0, button="runtool_btn", form=None, **kwd):
         # SubmitControl, ImageControl
         if form is None:
-            for i, form in enumerate(self.showforms()):
+            for i, _form in enumerate(self.showforms()):
```
Here `form` is used after the for loop, so a different workaround is needed, depending on what is returned by `self.showforms()`, e.g. add `form = _form` right before the `break` at line 181.

nsoranzo committed:
Save index as section_in_elem_list

nsoranzo committed:
Restore enumerate

nsoranzo committed:
Remove unused exception variable

nsoranzo committed:
Fix typo

bgruening committed:
fix some loop variable

bgruening committed:
add todo to some strange code

bgruening committed:
merge

bgruening committed:
add asserts

bgruening committed:
remove more enumerates

bgruening committed:
use better var name

bgruening committed:
remove enumerate()

bgruening committed:
fix pep8

bgruening committed:
fix whitespaces

nsoranzo commented on lib/galaxy_test/selenium/framework.py:
```diff
@@ -134,7 +134,7 @@ def write_file(name, content, raw=False):
             try:
                 full_log = self.driver.get_log(log_type)
-                trimmed_log = [l for l in full_log if l["level"] not in ["DEBUG", "INFO"]]
+                trimmed_log = [log for log in full_log if log["level"] not in ["DEBUG", "INFO"]]
```
```suggestion
                trimmed_log = [entry for entry in full_log if entry["level"] not in ["DEBUG", "INFO"]]
```

nsoranzo commented on lib/tool_shed/test/base/twilltestcase.py:
```diff
@@ -174,8 +176,9 @@ def submit_form(self, form_no=0, button="runtool_btn", form=None, **kwd):
                 if i == form_no:
+                    form = _form
                     break
         assert form, "No form to submit found"
```
```suggestion
            try:
                form = self.showforms()[form_no]
            except IndexError:
                raise ValueError("No form to submit found")
```

nsoranzo commented on lib/galaxy/tool_shed/tools/data_table_manager.py:
```diff
@@ -145,7 +145,7 @@ def install_tool_data_tables(self, tool_shed_repository, tool_index_sample_files
                     # Append individual table elems or other elemes, but not tables elems.
                     if elem.tag == 'tables':
-                        for table_elem in elems:
+                        for _table_elem in elems:
```
```suggestion
                        # TODO: this code need to be revised
                        for _table_elem in elems:
```

bgruening committed:
Update lib/galaxy_test/selenium/framework.py

Co-authored-by: Nicola Soranzo <nicola.soranzo@earlham.ac.uk>

bgruening committed:
Update lib/galaxy/tool_shed/tools/data_table_manager.py

Co-authored-by: Nicola Soranzo <nicola.soranzo@earlham.ac.uk>

bgruening committed:
Update lib/tool_shed/test/base/twilltestcase.py

Co-authored-by: Nicola Soranzo <nicola.soranzo@earlham.ac.uk>

nsoranzo committed:
Fix `submit_form()` use

Previous code defaulted to the last form in the page if `form_no` did
not correspond to the index of an existing form.

Also remove unused `set_skip_tool_tsts_flag()` method.

nsoranzo committed:
Replace argument with [] only if None

For example a `ToolRequirements` argument of length 0 evaluates to
False and was replaced, causing the following obscure integration
test failures:

```
INFO     pulsar.managers.stateful:stateful.py:64 Status of job [4] changed to [failed]. No callbacks enabled.
ERROR    pulsar.managers.stateful:stateful.py:143 Failed job preprocessing for job 4:
Traceback (most recent call last):
  File "/home/runner/work/galaxy/galaxy/galaxy root/.venv/lib/python3.7/site-packages/pulsar/managers/stateful.py", line 132, in _handling_of_preprocessing_state
    **launch_kwds
  File "/home/runner/work/galaxy/galaxy/galaxy root/.venv/lib/python3.7/site-packages/pulsar/managers/queued.py", line 54, in launch
    setup_params=setup_params
  File "/home/runner/work/galaxy/galaxy/galaxy root/.venv/lib/python3.7/site-packages/pulsar/managers/unqueued.py", line 54, in _prepare_run
    setup_params=setup_params
  File "/home/runner/work/galaxy/galaxy/galaxy root/.venv/lib/python3.7/site-packages/pulsar/managers/base/directory.py", line 120, in _setup_job_file
    command_line = self._expand_command_line(command_line, dependencies_description, job_directory=self.job_directory(job_id).job_directory)
  File "/home/runner/work/galaxy/galaxy/galaxy root/.venv/lib/python3.7/site-packages/pulsar/managers/base/__init__.py", line 197, in _expand_command_line
    job_directory=job_directory,
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/tool_util/deps/__init__.py", line 185, in dependency_shell_commands
    requirements_to_dependencies = self.requirements_to_dependencies(requirements, **kwds)
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/tool_util/deps/__init__.py", line 195, in requirements_to_dependencies
    requirement_to_dependency = self._requirements_to_dependencies_dict(requirements, **kwds)
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/tool_util/deps/__init__.py", line 213, in _requirements_to_dependencies_dict
    resolvable_requirements = requirements.resolvable
AttributeError: 'list' object has no attribute 'resolvable'
```

Also rename variable name from `dict` to `d` to not use standard library reserved words.

jmchilton commented on lib/galaxy/tool_util/deps/brew_exts.py:
```diff
@@ -51,7 +51,7 @@ class BrewContext:
     def __init__(self, args=None):
```
this whole file should be axed... but next PR 😆. I think I removed the Planemo bits that used this.

jmchilton commented on lib/galaxy_test/api/test_workflows.py:
```diff
@@ -489,17 +489,17 @@ def test_refactor(self):
         refactor_response.raise_for_status()
         # it was with dry_run=True - so the result is unchanged...
-        refactor_response.json()["steps"]["0"]["label"] == "test_input"
+        assert refactor_response.json()["steps"]["0"]["label"] == "test_input"
```
Classic John

jmchilton reviewed (approved):
Very nice work, thanks so much!

bgruening commented on lib/galaxy/tool_util/deps/dependencies.py:
```diff
@@ -80,7 +82,11 @@ class DependenciesDescription:
 
-    def __init__(self, requirements=[], installed_tool_dependencies=[]):
+    def __init__(self, requirements=None, installed_tool_dependencies=None):
+        if requirements is None:
```
@nsoranzo do you know why this was here needed and not elsewhere?

nsoranzo replied:
@bgruening `requirements = requirements or []` would replace a `ToolRequirements` instance of length 0 with `[]`, which then fails when calling the `resolvable()` method on it. After investigating, I propose a better fix in https://github.com/galaxyproject/galaxy/pull/11044 .
